GPflow/kernels.py

```diff
@@ -105,7 +259,7 @@ class White(Static):
     The White kernel
     """
 
-    def K(self, X, X2=None):
+    def K(self, X, X2=None, presliced=False):
```

Comment:
When computing kernel expectations, it is nicer to pass in a pre-sliced `X`. This way, the quadrature can create the input points on a lower dimension, rather than compute points on many more dimensions, only for some to be discarded by the kernel function.

An alternative would be to create the quadrature grid, and then to reverse slice it to fill up the sliced away dimensions with dummy data (e.g. zeros). I thought this would be impractical to do with TensorFlow manipulations.